packages/auth/src/grant/routes.ts

```diff
@@ -102,6 +108,27 @@ async function createGrantInitiation(
     return
   }
 
+  if (
+    body.access_token.access
+      .map((acc) => {
+        return isIncomingPaymentAccessRequest(acc as IncomingPaymentRequest)
+      })
+      .every((el) => el === true) &&
+    !deps.config.incomingPaymentInteraction
```

Comment:
Check the boolean first
```suggestion
    !deps.config.incomingPaymentInteraction &&
    body.access_token.access
      .map((acc) => {
        return isIncomingPaymentAccessRequest(acc as IncomingPaymentRequest)
      })
      .every((el) => el === true)
```